src/main/java/com/mojang/serialization/codecs/EitherCodec.java

```diff
@@ -25,7 +25,11 @@ public EitherCodec(final Codec<F> first, final Codec<S> second) {
         if (firstRead.result().isPresent()) {
             return firstRead;
         }
-        return second.decode(ops, input).map(vo -> vo.mapFirst(Either::right));
+        final DataResult<Pair<Either<F, S>, T>> secondRead = second.decode(ops, input).map(vo -> vo.mapFirst(Either::right));
+        if (secondRead.result().isPresent()) {
+            return secondRead;
+        }
+        return firstRead.mapError(err -> secondRead.error().map(pr -> "Either [" + err + "; " + pr.message() + "]").orElse(err));
```

Comment:
You are changing the behaviour of the codec in the cases where both error. Right now the partial result of the 2nd type is returned, and here you are returning the first's.

Reply:
Nice catch. Should be fixable by switching the `firstRead` and `secondRead` order I think.